src/NServiceBus.Core/Hosting/Helpers/AssemblyScanningComponent.cs

```diff
@@ -31,8 +32,8 @@ public static AssemblyScanningComponent Initialize(Configuration configuration,
 
             var assemblyScannerSettings = configuration.AssemblyScannerConfiguration;
 
-            assemblyScanner.AssembliesToSkip = assemblyScannerSettings.ExcludedAssemblies;
-            assemblyScanner.TypesToSkip = assemblyScannerSettings.ExcludedTypes;
+            assemblyScanner.AssembliesToSkip = new HashSet<string>(assemblyScannerSettings.ExcludedAssemblies.Select(Path.GetFileNameWithoutExtension).ToArray(), StringComparer.OrdinalIgnoreCase);
```

Comment:
I don't like that this knowledge is outside but I figured since there is already tight coupling between the Component and the scanner and I wanted to fix some perf issues I'll leave it as is for now

Reply:
Then I would prefer the property accept the config type and convert it internally once and use the field within the code.